apigentools/commands/generate.py

```diff
@@ -315,6 +315,12 @@ def pull_repository(self, language):
                 ['git', 'clone', '--depth=2', {"item": repo, "secret": secret_repo_url}, output_dir],
                 sensitive_output=True
             )
+            if branch is not None:
+                try:
+                    run_command(['git', 'checkout', '--', branch], cwd=output_dir)
+                except subprocess.CalledProcessError:
+                    # if the branch doesn't exist, we stay in the default one
+                    run_command(['git', 'checkout', '-b', branch], cwd=output_dir)
```

Comment:
I don't get why we should do `git checkout -b <branch>` here. If the branch does not exist, we should just `pass`, shouldn't we?

Reply:
I would like to have a consistent behavior so when one uses the `--branch` option than she can expect to have the branch checkouted.

Another reason is that in the near future, this should act as a base branch for new PR.

If it doesn't convince you, then I remove it for now.

Reply:
Hmm, I'm not strictly against this change in naming of the branches that we push. That said, this will need to be accompanied by a change in the `push` command either way (IIUC in this case the `push` command should push the generated change to the branch that was checked out - right now, it wouldn't do that).

IOW, I'm ok with naming the branches according to that env variable all the time (even if said branch doesn't exist), but the `push` command needs to respect this.

Reply:
I will remove it for now and we can revisit when updating the push command.